tokio/src/coop.rs

```diff
@@ -98,6 +98,12 @@ where
     })
 }
 
+cfg_rt_threaded! {
+    pub(crate) fn has_budget_remaining() -> bool {
```

Comment:
You may want to `#[inline(always)]` this since it gets called in a pretty tight loop. It made a decently large difference for `coop::budget`.

Reply:
I’m surprised rustc doesn’t figure that out on its own!